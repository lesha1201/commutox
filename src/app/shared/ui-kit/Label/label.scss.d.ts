// This file is generated automatically
export const label: string;
