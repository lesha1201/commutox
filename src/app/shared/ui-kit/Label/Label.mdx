---
name: Label
menu: UI Kit
---

import { Playground, PropsTable } from 'docz';
import Margin from 'src/app/shared/docz-helpers/Margin';
import Input from 'src/app/shared/ui-kit/Input';
import Label from './index';

# Label

\<Label \/> is just html tag \<label \/> with styles.

## Props

<PropsTable of={Label} />

## Default

<Playground>
  <div style={{ fontSize: '14px' }}>
    <Label htmlFor="first-name" style={{ marginBottom: '6px' }}>
      First name
    </Label>
    <Input id="first-name" placeholder="First name" width="full" />
  </div>
</Playground>
