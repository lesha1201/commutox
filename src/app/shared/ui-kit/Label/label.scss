@import 'app/shared/ui-kit/sass/variables';

.label {
  display: inline-block;

  color: $color-primary-black;

  font-size: 1.2em;
  font-family: inherit;
}
