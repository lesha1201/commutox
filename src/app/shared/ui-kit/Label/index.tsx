import * as React from 'react';

import * as style from './label.scss';

export interface IProps extends React.LabelHTMLAttributes<HTMLLabelElement> {}

const Label: React.FC<IProps> = props => {
  return <label className={style.label} {...props} />;
};

export default Label;
