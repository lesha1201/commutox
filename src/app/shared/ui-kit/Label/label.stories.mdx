import { Meta, Story, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';

import { Margin } from 'app/shared/storybook-helpers';
import { Input } from 'app/shared/ui-kit';
import Label from './index';

# Label

<Meta title="UI Kit/Label" component={Label} />

`<Label />` is just html tag `<label />` with styles.

## Stories

### Default

<Canvas withToolbar>
  <Story name="default">
    <div style={{ fontSize: '14px' }}>
      <Label htmlFor="first-name" style={{ marginBottom: '6px' }}>
        First name
      </Label>
      <Input id="first-name" placeholder="First name" width="full" />
    </div>
  </Story>
</Canvas>

## Props

<ArgsTable of={Label} />
