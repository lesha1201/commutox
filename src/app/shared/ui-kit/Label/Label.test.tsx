import * as React from 'react';
import { render } from 'react-testing-library';

import Label from '.';

describe('<Label />', () => {
  it('should render', () => {
    const { container } = render(<Label htmlFor="id" style={{ fontSize: '20px' }} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
