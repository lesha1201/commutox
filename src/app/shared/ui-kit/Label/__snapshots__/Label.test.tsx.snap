// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render 1`] = `
<label
  class="label"
  for="id"
  style="font-size: 20px;"
/>
`;
